docs/notebooks/flax_basics.ipynb

```diff
@@ -0,0 +1,911 @@
+{
```

Comment:
<p>In the spirit of making things shorter : You are mentioning here that you will pass a PRNGKey and dummy input parameters, then you have the code cell where you pass the PRNGKey and dummy input parameters (where you also added a comment that it is a dummy input), and then you have a long explanation afterwards why you do this.</p><p> </p><p>In general I think it would be great if the text could be made shorter. Long text cells significantly slow down the progress through the notebook and I suspect that fewer people finish it if they have to read too much (we're living in a information-supersaturated world and uninterrupted focus is a scarce resource ;-)</p><p> </p><p> </p><p>An extreme version would be something like this:</p><p> </p><p>Text : Note that the module does not contain it's parameters, but instead they first have to be generated by calling <code>.init()</code> :</p><p> </p><pre class="ql-syntax" spellcheck="false">key0 = random.PRNGKey(0) # Random seed for weight initialization.
params = model.init(key0, jnp.zeros(10)) # Shape inference from fake input.
jax.tree_map(jnp.shape, params)
</pre><p> </p><p>Text : (talking about the next cell - when combining text &amp; code cells I find it useful to have a clear relationship between text and code cells, e.g. always having a text talking about the next code cell, but YMMV)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google/flax/pull/542/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Reply:
<p>I generally agree with the points being made here, and I'll definitely revisit the test to make it shorter and straighter to the point. I wrote this in one go, so I'm not surprised it's not "length-optimized" but every point in the text is still useful I think because I added what I felt I was missing during the learning process. I'll address other comments and then rework on the text.</p><p> </p><p>About text talking about the next cell, I think this is generally what I try to do, and when I don't it's because I need the output of the previous cell. I'll revisit that as well when rewriting.</p>